"pkg-config --exists libffi"
package configuration for libffi is not found
have_header: checking for ffi.h... -------------------- no

"clang -o conftest -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0/x86_64-darwin18 -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0/ruby/backward -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0 -I. -I/Users/kei/.rbenv/versions/2.6.5/include  -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT    -O3 -Wno-error=shorten-64-to-32  -fno-common -pipe conftest.c  -L. -L/Users/kei/.rbenv/versions/2.6.5/lib -L. -L/Users/kei/.rbenv/versions/2.6.5/lib  -fstack-protector-strong -L/usr/local/lib     -lruby.2.6   "
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: int main(int argc, char **argv)
4: {
5:   return 0;
6: }
/* end */

"clang -E -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0/x86_64-darwin18 -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0/ruby/backward -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0 -I. -I/Users/kei/.rbenv/versions/2.6.5/include  -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT    -O3 -Wno-error=shorten-64-to-32  -fno-common -pipe  conftest.c -o conftest.i"
conftest.c:3:10: fatal error: 'ffi.h' file not found
#include <ffi.h>
         ^~~~~~~
1 error generated.
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <ffi.h>
/* end */

--------------------

find_header: checking for ffi.h in /usr/local/include,/usr/include/ffi,/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/ffi,/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/ffi... -------------------- yes

"clang -E -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0/x86_64-darwin18 -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0/ruby/backward -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0 -I. -I/Users/kei/.rbenv/versions/2.6.5/include  -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT    -O3 -Wno-error=shorten-64-to-32  -fno-common -pipe  conftest.c -o conftest.i"
conftest.c:3:10: fatal error: 'ffi.h' file not found
#include <ffi.h>
         ^~~~~~~
1 error generated.
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <ffi.h>
/* end */

"clang -E -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0/x86_64-darwin18 -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0/ruby/backward -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0 -I. -I/Users/kei/.rbenv/versions/2.6.5/include  -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT    -O3 -Wno-error=shorten-64-to-32  -fno-common -pipe -I/usr/local/include conftest.c -o conftest.i"
conftest.c:3:10: fatal error: 'ffi.h' file not found
#include <ffi.h>
         ^~~~~~~
1 error generated.
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <ffi.h>
/* end */

"clang -E -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0/x86_64-darwin18 -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0/ruby/backward -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0 -I. -I/Users/kei/.rbenv/versions/2.6.5/include  -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT    -O3 -Wno-error=shorten-64-to-32  -fno-common -pipe -I/usr/include/ffi conftest.c -o conftest.i"
conftest.c:3:10: fatal error: 'ffi.h' file not found
#include <ffi.h>
         ^~~~~~~
1 error generated.
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <ffi.h>
/* end */

"clang -E -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0/x86_64-darwin18 -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0/ruby/backward -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0 -I. -I/Users/kei/.rbenv/versions/2.6.5/include  -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT    -O3 -Wno-error=shorten-64-to-32  -fno-common -pipe -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/ffi conftest.c -o conftest.i"
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <ffi.h>
/* end */

--------------------

have_library: checking for ffi_prep_closure_loc() in -lffi... -------------------- no

"clang -o conftest -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0/x86_64-darwin18 -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0/ruby/backward -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0 -I. -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/ffi -I/Users/kei/.rbenv/versions/2.6.5/include  -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT    -O3 -Wno-error=shorten-64-to-32  -fno-common -pipe conftest.c  -L. -L/Users/kei/.rbenv/versions/2.6.5/lib -L. -L/Users/kei/.rbenv/versions/2.6.5/lib  -fstack-protector-strong -L/usr/local/lib     -lruby.2.6 -lffi   "
conftest.c:16:57: error: use of undeclared identifier 'ffi_prep_closure_loc'
int t(void) { void ((*volatile p)()); p = (void ((*)()))ffi_prep_closure_loc; return !p; }
                                                        ^
1 error generated.
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: #include <ffi.h>
 4: 
 5: /*top*/
 6: extern int t(void);
 7: int main(int argc, char **argv)
 8: {
 9:   if (argc > 1000000) {
10:     int (* volatile tp)(void)=(int (*)(void))&t;
11:     printf("%d", (*tp)());
12:   }
13: 
14:   return 0;
15: }
16: int t(void) { void ((*volatile p)()); p = (void ((*)()))ffi_prep_closure_loc; return !p; }
/* end */

"clang -o conftest -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0/x86_64-darwin18 -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0/ruby/backward -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0 -I. -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/ffi -I/Users/kei/.rbenv/versions/2.6.5/include  -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT    -O3 -Wno-error=shorten-64-to-32  -fno-common -pipe conftest.c  -L. -L/Users/kei/.rbenv/versions/2.6.5/lib -L. -L/Users/kei/.rbenv/versions/2.6.5/lib  -fstack-protector-strong -L/usr/local/lib     -lruby.2.6 -lffi   "
Undefined symbols for architecture x86_64:
  "_ffi_prep_closure_loc", referenced from:
      _t in conftest-259727.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: #include <ffi.h>
 4: 
 5: /*top*/
 6: extern int t(void);
 7: int main(int argc, char **argv)
 8: {
 9:   if (argc > 1000000) {
10:     int (* volatile tp)(void)=(int (*)(void))&t;
11:     printf("%d", (*tp)());
12:   }
13: 
14:   return 0;
15: }
16: extern void ffi_prep_closure_loc();
17: int t(void) { ffi_prep_closure_loc(); return 0; }
/* end */

--------------------

have_library: checking for ffi_prep_closure_loc() in -llibffi... -------------------- no

"clang -o conftest -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0/x86_64-darwin18 -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0/ruby/backward -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0 -I. -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/ffi -I/Users/kei/.rbenv/versions/2.6.5/include  -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT    -O3 -Wno-error=shorten-64-to-32  -fno-common -pipe conftest.c  -L. -L/Users/kei/.rbenv/versions/2.6.5/lib -L. -L/Users/kei/.rbenv/versions/2.6.5/lib  -fstack-protector-strong -L/usr/local/lib     -lruby.2.6 -llibffi   "
conftest.c:16:57: error: use of undeclared identifier 'ffi_prep_closure_loc'
int t(void) { void ((*volatile p)()); p = (void ((*)()))ffi_prep_closure_loc; return !p; }
                                                        ^
1 error generated.
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: #include <ffi.h>
 4: 
 5: /*top*/
 6: extern int t(void);
 7: int main(int argc, char **argv)
 8: {
 9:   if (argc > 1000000) {
10:     int (* volatile tp)(void)=(int (*)(void))&t;
11:     printf("%d", (*tp)());
12:   }
13: 
14:   return 0;
15: }
16: int t(void) { void ((*volatile p)()); p = (void ((*)()))ffi_prep_closure_loc; return !p; }
/* end */

"clang -o conftest -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0/x86_64-darwin18 -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0/ruby/backward -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0 -I. -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/ffi -I/Users/kei/.rbenv/versions/2.6.5/include  -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT    -O3 -Wno-error=shorten-64-to-32  -fno-common -pipe conftest.c  -L. -L/Users/kei/.rbenv/versions/2.6.5/lib -L. -L/Users/kei/.rbenv/versions/2.6.5/lib  -fstack-protector-strong -L/usr/local/lib     -lruby.2.6 -llibffi   "
ld: library not found for -llibffi
clang: error: linker command failed with exit code 1 (use -v to see invocation)
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: #include <ffi.h>
 4: 
 5: /*top*/
 6: extern int t(void);
 7: int main(int argc, char **argv)
 8: {
 9:   if (argc > 1000000) {
10:     int (* volatile tp)(void)=(int (*)(void))&t;
11:     printf("%d", (*tp)());
12:   }
13: 
14:   return 0;
15: }
16: extern void ffi_prep_closure_loc();
17: int t(void) { ffi_prep_closure_loc(); return 0; }
/* end */

--------------------

have_library: checking for ffi_prep_closure_loc() in -llibffi-8... -------------------- no

"clang -o conftest -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0/x86_64-darwin18 -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0/ruby/backward -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0 -I. -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/ffi -I/Users/kei/.rbenv/versions/2.6.5/include  -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT    -O3 -Wno-error=shorten-64-to-32  -fno-common -pipe conftest.c  -L. -L/Users/kei/.rbenv/versions/2.6.5/lib -L. -L/Users/kei/.rbenv/versions/2.6.5/lib  -fstack-protector-strong -L/usr/local/lib     -lruby.2.6 -llibffi-8   "
conftest.c:16:57: error: use of undeclared identifier 'ffi_prep_closure_loc'
int t(void) { void ((*volatile p)()); p = (void ((*)()))ffi_prep_closure_loc; return !p; }
                                                        ^
1 error generated.
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: #include <ffi.h>
 4: 
 5: /*top*/
 6: extern int t(void);
 7: int main(int argc, char **argv)
 8: {
 9:   if (argc > 1000000) {
10:     int (* volatile tp)(void)=(int (*)(void))&t;
11:     printf("%d", (*tp)());
12:   }
13: 
14:   return 0;
15: }
16: int t(void) { void ((*volatile p)()); p = (void ((*)()))ffi_prep_closure_loc; return !p; }
/* end */

"clang -o conftest -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0/x86_64-darwin18 -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0/ruby/backward -I/Users/kei/.rbenv/versions/2.6.5/include/ruby-2.6.0 -I. -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/ffi -I/Users/kei/.rbenv/versions/2.6.5/include  -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT    -O3 -Wno-error=shorten-64-to-32  -fno-common -pipe conftest.c  -L. -L/Users/kei/.rbenv/versions/2.6.5/lib -L. -L/Users/kei/.rbenv/versions/2.6.5/lib  -fstack-protector-strong -L/usr/local/lib     -lruby.2.6 -llibffi-8   "
ld: library not found for -llibffi-8
clang: error: linker command failed with exit code 1 (use -v to see invocation)
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: #include <ffi.h>
 4: 
 5: /*top*/
 6: extern int t(void);
 7: int main(int argc, char **argv)
 8: {
 9:   if (argc > 1000000) {
10:     int (* volatile tp)(void)=(int (*)(void))&t;
11:     printf("%d", (*tp)());
12:   }
13: 
14:   return 0;
15: }
16: extern void ffi_prep_closure_loc();
17: int t(void) { ffi_prep_closure_loc(); return 0; }
/* end */

--------------------

extconf.h is:
/* begin */
1: #ifndef EXTCONF_H
2: #define EXTCONF_H
3: #define HAVE_FFI_PREP_CIF_VAR 1
4: #define USE_INTERNAL_LIBFFI 1
5: #endif
/* end */

